src/index.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  Beautifier,
+  Language,
+  BeautifierBeautifyData,
+  DependencyType,
+  ExecutableDependency,
+} from "unibeautify";
+import * as readPkgUp from "read-pkg-up";
+import * as tmp from "tmp";
+import * as fs from "fs";
+import * as path from "path";
+import * as os from "os";
+
+const { pkg } = readPkgUp.sync({ cwd: __dirname });
+
+export const beautifier: Beautifier = {
+  name: "clang-format",
+  package: pkg,
+  badges: [
+    {
+      description: "Build Status",
+      url:
+        "https://travis-ci.com/Unibeautify/beautifier-clang-format.svg?branch=master",
+      href: "https://travis-ci.com/Unibeautify/beautifier-clang-format",
+    },
+    {
+      description: "Test Coverage",
+      url:
+        "https://api.codeclimate.com/v1/badges/8b4e5b19de96f4744ffa/test_coverage",
+      href:
+        "https://codeclimate.com/github/Unibeautify/beautifier-clang-format/test_covera" +
+        "ge",
+    },
+    {
+      description: "Maintainability",
+      url:
+        "https://api.codeclimate.com/v1/badges/8b4e5b19de96f4744ffa/maintainability",
+      href:
+        "https://codeclimate.com/github/Unibeautify/beautifier-clang-format/maintainabi" +
+        "lity",
+    },
+    {
+      description: "Greenkeeper",
+      url:
+        "https://badges.greenkeeper.io/Unibeautify/beautifier-clang-format.svg",
+      href: "https://greenkeeper.io/",
+    },
+  ],
+  options: {
+    "C": true,
+    "C++": true,
+    Java: true,
+    "Objective-C": true
+  },
+  dependencies: [
+    {
+      type: DependencyType.Executable,
+      name: "clang-format",
+      program: "clang-format",
+      parseVersion: [/clang-format version (\d+\.\d+\.\d+)/],
+      homepageUrl: "https://clang.llvm.org/docs/ClangFormat.html",
+      installationUrl: "https://clang.llvm.org/docs/ClangFormat.html",
+      bugsUrl: "https://clang.llvm.org/docs/ClangFormat.html",
+      badges: [
+        {
+          description: "",
+          url: "",
+          href: "",
+        },
+      ],
+    },
+  ],
+  resolveConfig: ({ filePath, projectPath }) => {
+    const configFiles: string[] = [".clang-format", "_clang-format"];
```

Comment:
What're your thoughts on making `configFiles` a more first class feature of the beautifier?

```diff
-  resolveConfig: ({ filePath, projectPath }) => {
- // .... lots of code
- }
+ resolveConfigFiles: [".clang-format", "_clang-format"],
```

This provides:
1) a shorthand for this code we've been reusing
2) allows us to automatically generate documentation on website

Also, what about configuration examples or something else we could use to improve the website documentation for usage of `prefer beautifier config`.

Reply:
I agree with that!